libtextsecure/sendmessage.js

```diff
@@ -217,7 +222,10 @@ MessageSender.prototype = {
       attachmentData = result.ciphertext;
     }
 
-    const result = await server.putAttachment(attachmentData);
+    const result = isAvatar
+      ? await server.putAvatar(attachmentData)
+      : await server.putAttachment(attachmentData);
```

Comment:
Would `await server.putAvatar(attachmentData)` possibly only return `{ url }` but not an id? is that ok?

Reply:
Looks like that would be fine, he's just checking to make sure both don't return a falsish value (failed uploading).

Reply:
I'm not using the `id` anywhere for avatars (should I?) and the rest of the code still works, so I omitted the field.

Reply:
My question was related to those lines below:
```js
const { url, id } = result;
proto.id = id;
```
We're setting `undefined` to `proto.id` but we're not using `proto.id` right?

Reply:
Correct, `proto.id` will be `undefined`.